Feature/points

patryklomza opened the pull request:
This is a starting point for our Points system.

Check updated README for a brief endpoints description.

I will add an endpoint calculating total student points in a later PR. 

Waiting for your questions, and suggestions on what to improve!

patryklomza committed:
Add POST request test as first test

patryklomza committed:
Move fixtures directory to users app

Moved fixtures so that we could use it also in pytest tests.

patryklomza committed:
Update Point model with auto-adding date and time

Now Point instance has date and time of its creation.

patryklomza committed:
Add Point serializer

patryklomza committed:
Add Point APIView resource and url handling

patryklomza committed:
Add pytest configuration with fixtures loading

Now for each session of tests, test database gets populated with the
data from fixtures.

patryklomza committed:
Add tests for invalid payload

patryklomza committed:
Add invalid payload handling

patryklomza committed:
Update test for invalid payload of POST request

patryklomza committed:
Add tests checking GET request for single point

patryklomza committed:
Add single point APIView

patryklomza committed:
Add test for invalid point identifier

patryklomza committed:
Add test for getting points of a given student

patryklomza committed:
Add list APIView for points of a given student

patryklomza committed:
Update tested endpoints in Point tests

Flat resources structure should ease API usage.

patryklomza committed:
Remove redundant urls and imports

patryklomza committed:
Refactor Point related resources to one resource

patryklomza committed:
Refactor Point related urls

patryklomza committed:
Add missing pytest dependency

patryklomza committed:
Fix faulty false positive test

patryklomza committed:
Fix handling of a GET points/ endpoint

patryklomza committed:
Add missing whitespace at the end of the file

patryklomza committed:
Add test for invalid student query string

patryklomza committed:
Refactor GET handling in Point resource

patryklomza committed:
Add 'pk' field to the Point serializer

patryklomza committed:
Add env variable for django settings module

patryklomza committed:
Move point tests module to the proper directory

patryklomza committed:
Update README.md

visse0001 commented:
Could you push 'makemigrations'?

Nice README description. 
And the tests! Too bad when I run them I get " no tests ran in 0.09s" :thinking: 
After hitting endpoint /api/users/points/?studentId=2 I have: 
```
"AssertionError at /api/users/points/
'PointResource' should either include a `serializer_class` attribute, or override the `get_serializer_class()` method."
```

visse0001 commented:
Could you move `pytest.ini` and tests to root dir? ;)

patryklomza committed:
Move fixture and test directories to project root

Also removed a DJANGO_SETTINGS_MODULE environment variable as it is not
necessary with the new directory composition.

patryklomza committed:
Add migration script

patryklomza committed:
Add FIXTURE_DIRS setting

patryklomza commented:
> Could you move `pytest.ini` and tests to root dir? ;)

✅ done

patryklomza commented:
> Could you push 'makemigrations'?
> 
> Nice README description.
> And the tests! Too bad when I run them I get " no tests ran in 0.09s" 🤔
> After hitting endpoint /api/users/points/?studentId=2 I have:
> 
> ```
> "AssertionError at /api/users/points/
> 'PointResource' should either include a `serializer_class` attribute, or override the `get_serializer_class()` method."
> ```

✅ done

visse0001 commented on users/resources.py:
```diff
@@ -109,3 +110,35 @@ def patch(self, request, pk):
         return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+class PointResource(generics.GenericAPIView):
```
Looks good. Could you add the serializer_class attribute here? and let’s merge this :)

patryklomza replied:
✅ done

patryklomza committed:
Add missing serializer_class to PointResource

patryklomza committed:
Fix poorly overwritten get_queryset method

visse0001 commented on README.md:
```diff
@@ -167,3 +167,35 @@ Go to
+There are currently several enpoints responsible for managing student points available:
+
+1. `api/users/points/` accesible with a `GET` request and followed by a query string pointing to a student primary key.  
+For example: `api/users/points/?studentId=2/` should return Point instances assigned to the student with `pk=2`  
```
`api/users/points/?studentId=2/` this trailing slash is redundant and couses ValueError :)

patryklomza replied:
✅ done

patryklomza committed:
Remove redundant trailing slash in url example
